### PR TITLE
rgw: dump s3_code as the Code response element in RGWDeleteMultiObj_ObjStore_S3

### PR DESCRIPTION
**Pull request by rzarzynski**
http://tracker.ceph.com/issues/18241

### CODE REVIEW

**Comment by Liuchang0812** [comment]
@rzarzynski @yehudasa I have submitted another PR about this #12485 

I would like to fix this bug, and add some test cases. How about?

**Comment by rzarzynski** [comment]
@Liuchang0812: I'm completely fine if you want to take care of this bug. :-) Just please be aware there is [an ongoing rework](https://github.com/ceph/ceph/pull/10690) in the area of error handling.

**Comment by Liuchang0812** [comment]
@rzarzynski thanks. I will take care of the ongoing rework. I am very pleasure to take part in this work.

**Comment by Liuchang0812** [comment]
@rzarzynski I closed #12485 . This PR could be ongoing. very thanks

File: src/rgw/rgw_rest_s3.cc [hunk 1]
```diff
@@ -2882,7 +2882,7 @@ void RGWDeleteMultiObj_ObjStore_S3::send_partial_response(rgw_obj_key& key,
 
       s->formatter->dump_string("Key", key.name);
       s->formatter->dump_string("VersionId", key.instance);
-      s->formatter->dump_int("Code", r.http_ret);
+      s->formatter->dump_string("Code", r.s3_code);
```
Comment (mattbenjamin) [inline src/rgw/rgw_rest_s3.cc]: Hi Amit,

Sometimes exceeding 50 characters in a PR title (or bz subject) is probably justified.  That's something that the RGW reviewers would probably flag, if it were a problem, in the case of an RGW PR.

cheers,

Matt

Matt

Comment (amitkumar50) [inline src/rgw/rgw_rest_s3.cc]: @mattbenjamin Thanks. 
Was just an friendly reminder if developer skipped at time of coding.

**Comment by rzarzynski** [comment]
jenkins retest this please (no log)

**Comment by tchaikov** [comment]
@yuriw could you include this change in your next batch?

**Comment by cbodley** [comment]
passed qa in http://pulpito.ceph.com/cbodley-2019-05-30_18:03:29-rgw-wip-cbodley-testing-distro-basic-smithi/

**Pull request merged**